device_doctor/bin/main.dart

```diff
@@ -56,3 +74,23 @@ Future<void> main(List<String> args) async {
       break;
   }
 }
+
+/// Different properties need to be defined for corresponding devices:
+///
+///   ios: `idevice_id`, `idevicediagnostics`
+///   android: `adb`
+void _validateProperties(String deviceOS, Map<String, dynamic> properties) {
```

Comment:
Properties looks like an implementation detail that should be in the device class, not the CLI tooling.

Reply:
Moved to device.